frontend/src/metabase/admin/performance/hooks/useConfirmIfFormIsDirty.tsx

```diff
@@ -0,0 +1,44 @@
+import { useState, useCallback } from "react";
+import type { InjectedRouter, Route } from "react-router";
+import { t } from "ttag";
+
+import useBeforeUnload from "metabase/hooks/use-before-unload";
+import { useConfirmation } from "metabase/hooks/use-confirmation";
+
+import { useConfirmOnRouteLeave } from "./useConfirmOnRouteLeave";
+
+export const useConfirmIfFormIsDirty = (
+  router: InjectedRouter,
+  route?: Route,
+) => {
+  const [isStrategyFormDirty, setIsStrategyFormDirty] = useState(false);
```

Comment:
could this component be more generally useful if if this was just `isFormDirty` ?